File: src/com/irisine/jiris/compiler/assistpart/IrisTopExpressionApostropheForeNormalCall.java
package com.irisine.jiris.compiler.assistpart;

import com.irisine.jiris.compiler.expression.IrisExpression;
import com.irisine.jiris.compiler.expression.IrisFunctionCallExpression;

import java.util.LinkedList;

/**
 * Created by Huisama on 2017/4/17 0017.
 */
public class IrisTopExpressionApostropheForeNormalCall extends IrisTopExpressionApostropheBase {

    private IrisIdentifier m_functionName = null;
    private LinkedList<IrisExpression> m_parameters = null;
    private IrisClosureBlockLiteral m_closureBlockLiteral = null;

    public IrisTopExpressionApostropheForeNormalCall(IrisIdentifier functionName, LinkedList<IrisExpression>
            prameters, IrisClosureBlockLiteral closureBlockLiteral) {
        m_functionName = functionName;
        m_parameters = prameters;
        m_closureBlockLiteral = closureBlockLiteral;
    }

    @Override
    public IrisExpression ToDirectExpression(IrisExpression foreExpression, IrisTopExpressionApostrophe postExpression) {
        IrisExpression tmpExpr = new IrisFunctionCallExpression(foreExpression, m_functionName, m_parameters, m_closureBlockLiteral);
        return postExpression == null ? tmpExpr : postExpression.ToDirectExpression(tmpExpr);
    }
}

File: src/main/java/com/irisine/jiris/compiler/assistpart/IrisTopExpressionApostropheForeNormalCall.kt
package com.irisine.jiris.compiler.assistpart

import com.irisine.jiris.compiler.expression.IrisExpression
import com.irisine.jiris.compiler.expression.IrisFunctionCallExpression

import java.util.LinkedList

/**
 * Created by Huisama on 2017/4/17 0017.
 */
class IrisTopExpressionApostropheForeNormalCall(val functionName: IrisIdentifier,
                                                val parameters: LinkedList<IrisExpression>?,
                                                val closureBlockLiteral: IrisClosureBlockLiteral?)
    : IrisTopExpressionApostropheBase() {

    override fun ToDirectExpression(foreExpression: IrisExpression, postExpression: IrisTopExpressionApostrophe?): IrisExpression {
        val tmpExpr = IrisFunctionCallExpression(foreExpression, functionName, parameters, closureBlockLiteral)
        return if (postExpression == null) tmpExpr else postExpression.ToDirectExpression(tmpExpr)
    }
}
